src/main/java/catbot/Parser.java

```diff
@@ -0,0 +1,80 @@
+package catbot;
+
+import catbot.internal.NamedParameterMap;
+
+import java.util.Arrays;
+
+public class Parser {
+
+    private final String delimiter;
+    private final boolean emptyArgument;
+
+    //region Constructor
+
+    private Parser(String delimiter, boolean emptyArgument) {
+        this.delimiter = delimiter;
+        this.emptyArgument = emptyArgument;
+    }
+
+    public static Parser by(String delimiter) {
+        return by(delimiter, false);
+    }
+
+    public static Parser by(String delimiter, boolean emptyArgument) {
```

Comment:
Perhaps adding javadocs for public functions might help explain their functioning in greater detail?